main.tf

```diff
@@ -56,7 +54,11 @@ resource "ibm_en_destination_cos" "cos_en_destination" {
 #############################################################################
 
 locals {
-  en_integration_id = length(data.ibm_en_integrations.en_integrations) > 0 ? data.ibm_en_integrations.en_integrations[0].integrations[0]["id"] : null
+
+  en_integration_id = length(data.ibm_en_integrations.en_integrations) > 0 ? [
+    for integrations in data.ibm_en_integrations.en_integrations[0].integrations :
+    integrations.id if(integrations.type == "kms" || integrations.type == "hs-crypto")
```

Comment:
just checking, the ID is the same for KMS and HPCS?

Reply:
Yes by default EN provides same integration id for HPCS and KP and other integration id for COS integeration.